src/TimeTravel.elm

```diff
@@ -3,3 +3,104 @@ module TimeTravel exposing (addTimeTravel)
 import Playground exposing (..)
 import Set
 
+controlBarHeight = 64
+maxVisibleHistory = 2000
+
+addTimeTravel rawGame = 
+  { initialState = initialStateWithTimeTravel rawGame
+  , updateState = updateWithTimeTravel rawGame
+  , view = viewWithTimeTravel rawGame
+  }
+
+initialStateWithTimeTravel rawGame = 
+  { rawModel = rawGame.initialState
+  , paused = False
+  , history = []
+  , historyPlaybackPosition = 0
+  }
+
+updateWithTimeTravel rawGame computer currentModel =
+  let
```

Comment:
I decided to put that let block outside of the if statement so that I could have this updatedModel shorthand. which itself resulted from a desire to have the first new model after resuming actually be updated. Most of it is pretty similar.